src/components/ui/RadioSelect.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="radio-select">
+    <div
+      class="radio-select__option"
+      v-for="(radio, idx) in options"
+      :key="idx"
+    >
+      <input
+        type="checkbox"
```

Comment:
I see that it's some work but doing this with an actual radio button would be cool.
For now if fixable I'd suggest moving the `@click` to the div (which enables selecting by clicking anywhere on the div).

More important: it's not working with keyboard navigation. It might be cheaper to use radio buttons and style them (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) should work nicely with Vue.